### PR TITLE
raftstore: Avoid snapshot IO in raftstore thread

### PR DESCRIPTION
**Pull request by Connor1996**
### What is changed and how it works?

Issue Number: Close https://github.com/tikv/tikv/issues/16564

What's Changed:

```commit-message
Avoid snapshot IO in raftstore thread
```
This PR does:
- Only check snapshot meta file existence when receiving snapshot message so that don't need to read meta file content 
- Don't calculate checksum when loading snapshot meta, because it would check in `validate`

### Related changes

- [ ] PR to update `pingcap/docs`/`pingcap/docs-cn`:
- [x] Need to cherry-pick to the release branch

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [x] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

### Release note

```release-note
Avoid snapshot IO in raftstore thread
```

### CODE REVIEW

File: components/raftstore/src/store/snap.rs [hunk 1]
```diff
@@ -1710,6 +1700,20 @@ impl SnapManager {
         Ok(Box::new(s))
     }
 
+    pub fn check_snapshot_meta_file_exist(&self, key: &SnapKey) -> RaftStoreResult<()> {
```
Comment (LykxSassinator) [inline components/raftstore/src/store/snap.rs:1703]:
```suggestion
    pub fn meta_file_exists(&self, key: &SnapKey) -> RaftStoreResult<()> {
```

**Review by LykxSassinator — COMMENTED** [review]
Rest LGTM

**Review by glorv — APPROVED** [review]

**Review by SpadeA-Tang — APPROVED** [review]
LGTM. There's one unresolved comment, though.

File: components/raftstore/src/store/snap.rs [hunk 2]
```diff
@@ -1710,6 +1700,20 @@ impl SnapManager {
         Ok(Box::new(s))
     }
 
+    pub fn meta_file_exist(&self, key: &SnapKey) -> RaftStoreResult<()> {
+        let _lock = self.core.registry.rl();
+        let base = &self.core.base;
+        // Use CheckPolicy::None to avoid reading meta file
+        let s = Snapshot::new(base, key, false, CheckPolicy::None, &self.core)?;
+        if !file_exists(&s.meta_file.path) {
```
Comment (LykxSassinator) [inline components/raftstore/src/store/snap.rs:1708]:
```suggestion
        if !file_exists(s.meta_file.path) {
```

**Pull request merged**